packages/nightly-connect/src/lib/nightly-connect.ts

```diff
@@ -0,0 +1,223 @@
+        const tx = nearTransactions.Transaction.decode(Buffer.from(message));
+        const signedTx = await client!.signTransaction(tx);
+
+        return {
+          signature: signedTx.signature.data,
+          publicKey: tx.publicKey,
+        };
+      } catch (err) {
+        logger.log("Failed to sign message");
+        logger.error(err);
+
+        throw Error("Invalid message. Only transactions can be signed");
+      }
+    },
+  };
+  return {
+    async signIn() {
+      return new Promise((resolve, reject) => {
+        const existingAccounts = getAccounts();
+
+        if (existingAccounts.length) {
+          return resolve(existingAccounts);
+        }
+
+        try {
+          setupNightlyConnectState({
+            ...params,
+            icon: params.appIcon,
+            additionalInfo: params.additionalInfo || "",
+            onUserConnect: (pk) => {
+              connectedAccounts.push(pk);
+              modal.closeModal();
+              resolve(getAccounts());
+            },
+          }).then((appClient) => {
+            client = appClient;
+            modal.openModal(client.sessionId, Network.Near);
+          });
+        } catch (err) {
+          signOut();
+          reject(err);
+        }
+      });
+    },
+
+    signOut,
+
+    async getAccounts() {
+      return getAccounts().map(({ accountId }) => ({ accountId }));
+    },
+
+    async signAndSendTransaction({ signerId, receiverId, actions }) {
+      logger.log("signAndSendTransaction", { signerId, receiverId, actions });
+
+      const accounts = getAccounts();
+
+      if (!client?.sessionId || !accounts.length) {
+        throw new Error("Wallet not signed in");
+      }
+
+      const [signedTx] = await signTransactions(
+        transformTransactions([{ signerId, receiverId, actions }]),
+        signer,
+        options.network
+      );
+      return provider.sendTransaction(signedTx);
+    },
+
+    async signAndSendTransactions({ transactions }) {
+      logger.log("signAndSendTransactions", { transactions });
+
+      const accounts = getAccounts();
+
+      if (!client?.sessionId || !accounts.length) {
+        throw new Error("Wallet not signed in");
+      }
+
+      const signedTxs = await signTransactions(
+        transformTransactions(transactions),
+        signer,
+        options.network
+      );
+      return Promise.all(
+        signedTxs.map((signedTx) => provider.sendTransaction(signedTx))
+      );
+    },
+  };
+};
+
+export type SetupNightlyConnectParams = NightlyConnectParams & {
+  iconUrl?: string;
+};
+
+export function setupNightlyConnect({
+  additionalInfo,
+  application,
+  description,
+  timeout,
+  url,
+  appIcon,
+  iconUrl = "./assets/nightly-connect.png",
+}: SetupNightlyConnectParams): WalletModuleFactory<BridgeWallet> {
+  return async () => {
+    return {
+      id: "nightly-connect",
+      type: "bridge",
+      metadata: {
+        name: "NightlyConnect",
+        description: description,
```

Comment:
I think this is a mistake. `description` here is related the wallet (Nightly Connect) instead of the dApp if it's anything like WalletConnect?

Reply:
Yeah that's right. I removed it.